### ISSUE TITLE
Lot of errors in HowTo

### ISSUE THREAD

**Issue by senzacionale** [2020IP/TwentyTwenty.IdentityServer4.EntityFrameworkCore#17]
Hi, I would like to use your project but has many errors in HowTo section:

``` javascript
public void ConfigureServices(IServiceCollection services)
{
    ...
    services.AddEntityFrameworkSqlServer()
        .AddDbContext<ClientConfigurationContext>(o => o
            .UseSqlServer(connectionString, b =>
            b.MigrationsAssembly(typeof(Startup).GetTypeInfo().Assembly.GetName().Name)))
        .AddDbContext<ScopeConfigurationContext>(o => o
            .UseSqlServer((connectionString, , b =>
            b.MigrationsAssembly(typeof(Startup).GetTypeInfo().Assembly.GetName().Name)))
        .AddDbContext<OperationalContextEx>(o => o
            .UseSqlServer((connectionString, , b =>
            b.MigrationsAssembly(typeof(Startup).GetTypeInfo().Assembly.GetName().Name)));
    ...
}
```

and also

``` javascript
public void ConfigureServices(IServiceCollection services)
{
    ...
    var builder = services.AddIdentityServer(options =>
    {
        options.RequireSsl = false;
    });

    builder.ConfigureEntityFramework()
        .RegisterOperationalStores()
        .RegisterClientStore<Guid, ClientConfigurationContext>()
        .RegisterScopeStore<Guid, ScopeConfigurationContext<>();
    ...
}
```

can you please fix it and provide working HowTo's?

**Comment by ericgreenmix** [comment]
These are examples that do show the code needed to configure the use of this project. The code in the snippets above are correct, in what code you need to use this project. They will need to be formatted to specifically to fit your project.

**Comment by senzacionale** [comment]
Thank you for the reply. Code is syntactically wrong:

``` javascript
.RegisterScopeStore<Guid, ScopeConfigurationContext<>();
```

should be

``` javascript
.RegisterScopeStore<Guid, ScopeConfigurationContext>();
```

same as here, completely wrong. For example OperationalContextEx class does not exist in your project! connectionString, , b is without dot, etc ...

``` javascript
services.AddEntityFrameworkSqlServer()
        .AddDbContext<ClientConfigurationContext>(o => o
            .UseSqlServer(connectionString, b =>
            b.MigrationsAssembly(typeof(Startup).GetTypeInfo().Assembly.GetName().Name)))
        .AddDbContext<ScopeConfigurationContext>(o => o
            .UseSqlServer((connectionString, , b =>
            b.MigrationsAssembly(typeof(Startup).GetTypeInfo().Assembly.GetName().Name)))
        .AddDbContext<OperationalContextEx>(o => o
            .UseSqlServer((connectionString, , b =>
            b.MigrationsAssembly(typeof(Startup).GetTypeInfo().Assembly.GetName().Name)));
```

Can you provide syntactically correct HowTo? Or one simple working example.

I would like to migrate IdentityServer3 to version 4 and use your nuget for EF7...

regards,
Mitja

**Comment by ericgreenmix** [comment]
Ah, I see the typos now. I have fixed them in 711ab9d0e4f3e1cfe21ec7e5042cb78d46ee2feb. We currently do not have a working sample of Idsvr4 using this. We plan to add one in the future but it is not currently on our near term radar. #13 is tracking the sample project. Thanks.

**Comment by senzacionale** [comment]
thank you, but there is still one error, please add this to howto

```javascript
services.AddEntityFrameworkSqlServer()
                .AddDbContext<ClientConfigurationContext>(o => o
                    .UseSqlServer(connectionString, b =>
                    b.MigrationsAssembly(typeof(Startup).GetTypeInfo().Assembly.GetName().Name)))
                .AddDbContext<ScopeConfigurationContext>(o => o
                    .UseSqlServer(connectionString, b =>
                    b.MigrationsAssembly(typeof(Startup).GetTypeInfo().Assembly.GetName().Name)))
                .AddDbContext<OperationalContext>(o => o
                    .UseSqlServer(connectionString, b =>
                    b.MigrationsAssembly(typeof(Startup).GetTypeInfo().Assembly.GetName().Name)));
```

So if I add connectionString to my DB then everything should work out of the box?

This is ok, or I need something else here too?...

```javascript
public class ClientConfigurationContext : ClientConfigurationContext<Guid>
{
    public ClientConfigurationContext(DbContextOptions<ClientConfigurationContext> options)
        : base(options)
    { }
}
```
